plugin/src/Reconciler/init.lua

```diff
@@ -0,0 +1,34 @@
+--[[
+	This module defines the meat of the Rojo plugin and how it manages tracking
+	and mutating the Roblox DOM.
+]]
+
+local applyPatch = require(script.applyPatch)
+local hydrate = require(script.hydrate)
+local diff = require(script.diff)
+
+local Reconciler = {}
+Reconciler.__index = Reconciler
+
+function Reconciler.new(instanceMap)
+	local self = {
+		-- Tracks all of the instances known by the reconciler by ID.
+		__instanceMap = instanceMap,
+	}
+
+	return setmetatable(self, Reconciler)
+end
+
+function Reconciler:applyPatch(patch)
+	return applyPatch(self.__instanceMap, patch)
+end
+
+function Reconciler:hydrate(virtualInstances, rootId, rootInstance)
+	return hydrate(self.__instanceMap)
```

Comment:
Why are `virtualInstances`, `rootId` and `rootInstance` not passed to hydrate here?

Reply:
Good catch!